app/server/utils/ussd/kenya_ussd_processor.py

```diff
@@ -97,8 +97,69 @@ def custom_display_text(menu: UssdMenu, ussd_session: UssdSession, user: User) -
             else:
                 return i18n_for(user, "{}.first".format(menu.display_key))
 
-        return None
-
+        if menu.name == 'directory_listing' or menu.name == 'send_token_reason':
+            items_per_menu = 8
 
+            usages = user.get_most_relevant_transfer_usage()
```

Comment:
i think this makes sense to happen in the after in the state machine still. all setting of data would happen there, here it's just using stuff from the session data to get the display text

Reply:
mocking of the test would also be easier then, you just pass in data of your liking